### DISCUSSION TITLE
maintain task size independently of screen resolution - task coded with pixels coordinates

### DISCUSSION THREAD

**Discussion by EstelaCG** [jspsych/jsPsych#1030] [Q&A]
Hi all, 

I am coding a target cancellation task to present on a tablet/iPad, with a plugin that I am developing with Becky. The plugin allows the display of a target cancellation area div with distractors as well as targets all over the place to cancel when touched. A cancellation line appears on top of the target when this one is touched. The position of my 60 stimuli, as well as their corresponding feedback images, are based on pixels coordinates. Underneath that div, there is another div for a button to click when the task has been completed. The instructions are coded with an html-button-response-plugin. I am also presenting the task in full-screen mode. 

I am using a MacBook Pro 2020 (13.3-inch (2560x1600), and everything looks good, but I am aware that this might not be the case with other screens of different resolutions. I would like the size of my task to be the same in other devices. I have been playing around with the resize plugin, measuring a credit card, but it either makes the task too small or too big. Moreover, the buttons of the pages coded with the html-button-response-plugin appear over the instructions when resized. 

I find the resize documentation a bit confusin and I am not sure whether it doesn't work for me because I am doing something wrong or because this is not the solution that I am looking for. I have read that one solution would be not using pixel coordinates but inches or cm but ideally, I would prefer to avoid this as that would mean changing a lot of the code. Is there simpler way to do it? Thanks!

**Comment by jodeleeuw** [comment]
The resize plugin is meant to maintain the same physical size of objects on the screen, independent of screen resolution, i.e., no matter what screen is being used, if you took a ruler and measured the width of an image it would be the same.

Is that what you are after here? Or are you aiming to get the task to fill the same space on the monitor regardless of what kind of screen is being used, i.e., the objects appear bigger on a bigger screen, but have the same relative size and distance as on a smaller screen?

**Reply by EstelaCG** [reply]
Hi Josh,

Thank you for your prompt response. Yes, I want to maintain the same physical size, independent of screen resolution. However, I haven't been able to make it work. I am not sure I have understood the documentation correctly. If you could help me with that, it would be fabulous, thanks!

**Reply by jodeleeuw** [reply]
It occurred to me that the plugin may not work with high-def screens like Retina displays on the Mac. Those screens have atypical zooming settings enabled to account for the fact that the pixels are so tiny. Do you happen to have access to a different machine that you can test things on? I don't have access to a Retina display so I can't verify that this might be the cause of the problem.

**Reply by EstelaCG** [reply]
Hi Josh,

I tested the task with and without the resize plugin on a PC 13.3-inch (1920x1080). Both with and without the plugin, the task is bigger than the screen. I don't know what I am doing wrong. Here is a screenshot of the code for the plugin:

<img width="1063" alt="image001" src="https://user-images.githubusercontent.com/57395447/92518619-88fe7f00-f210-11ea-9f79-e0a9100971cf.png">


Thanks!

**Reply by jodeleeuw** [reply]
When you set `pixels_per_unit` to 100, this will make 1 inch = 100 pixels. Since your monitor is 1920 pixels wide, this will resize the content to be a maximum of 19.2 inches wide, which is larger than your screen. Try adjusting the value of `pixels_per_unit` to something smaller, until you find the size that you are looking for.

**Reply by EstelaCG** [reply]
Oh, now I understand, thanks! I did what you proposed, and it worked. However, the resize plugin doesn't work on an iPad - for some reason, it's impossible to drag the lower corner of the box using a touchscreen. Is there a solution for this? Thanks again.

**Reply by becky-gilbert** [reply]
Hi @EstelaCG! I was able to replicate this in Chrome by running the resize plugin example file and using the device emulator to switch to an iPad. I created an issue to track this (#1100).

I think the problem with using the resize plugin on touchscreens is that the plugin uses mousedown/mousemove/mouseup events, and apparently browsers will sometimes treat these as touch events on touchscreens, but not always. And [this posts](https://www.html5rocks.com/en/mobile/touchandmouse/) suggests that there are some important differences between mouse and touch events, so it may not work to just swap out the event names. But I'll have a look at it and see if there's a quick and easy way to make it touchscreen compatible.
